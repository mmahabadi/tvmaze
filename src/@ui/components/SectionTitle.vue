<script setup lang="ts">
defineProps<{ title: string }>()
</script>

<template>
  <div class="my-9">
    <div class="w-full flex sm:gap-8 gap-4 items-center">
      <div class="text-teal-500"><slot></slot></div>
      <h2 class="sm:text-xl font-bold text-lg">{{ title }}</h2>
    </div>
  </div>
</template>
