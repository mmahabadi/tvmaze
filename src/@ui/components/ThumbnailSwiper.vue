<script setup lang="ts">
import { SwiperSlide, Swiper } from 'swiper/vue'
import { Autoplay, Pagination, Navigation } from 'swiper/modules'
import { ref } from 'vue'
import ThumbnailItem from './ThumbnailItem.vue'
import { SlideItem } from '../types'

import 'swiper/css'
import 'swiper/css/autoplay'
import 'swiper/css/navigation'
import 'swiper/css/pagination'
import 'swiper/css/scrollbar'

defineProps<{ slides: SlideItem[] }>()

const modules = ref([Autoplay, Pagination, Navigation])
</script>
<template>
  <swiper
    :centeredSlides="true"
    :slides-per-view="5"
    :space-between="50"
    :navigation="true"
    :loop="true"
    :modules="modules"
    :scrollbar="{ draggable: true }"
  >
    <!-- :autoplay="{
      delay: 2500,
      disableOnInteraction: false
    }" -->
    <swiper-slide v-for="slide in slides" :key="slide.id">
      <div>
        <ThumbnailItem :item="slide" />
      </div>
    </swiper-slide>
  </swiper>
</template>
