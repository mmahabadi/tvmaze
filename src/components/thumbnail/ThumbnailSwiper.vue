<script setup lang="ts">
import { SwiperSlide } from 'swiper/vue'
import SwiperWrapper from '@/@ui/components/SwiperWrapper.vue'
import ThumbnailItem from './ThumbnailItem.vue'
import { SlideItem } from '@/types'
import { SwiperOptions } from '@/@ui/types'

defineProps<{ slides: SlideItem[] }>()

const options = {
  slidesPerView: 4,
  spaceBetween: 50,
  navigation: true,
  loop: true
} as SwiperOptions
</script>
<template>
  <SwiperWrapper v-bind="options">
    <swiper-slide v-for="slide in slides" :key="slide.id">
      <ThumbnailItem :item="slide" />
    </swiper-slide>
  </SwiperWrapper>
</template>
