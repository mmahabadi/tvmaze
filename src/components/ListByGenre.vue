<script setup lang="ts">
import { computed, defineProps, inject } from 'vue'
import { useMoviesStore } from '@/stores/movies'
import SectionTitle from '@/@ui/components/SectionTitle.vue'
import ThumbnailSwiper from '@/components/thumbnail/ThumbnailSwiper.vue'
import IconScreen from '@/components/icons/IconScreen.vue'
import type { AppSettings, SlideItem } from '@/types'
import { mapToSlideItems } from '@/utils/mapping'

const { genre } = defineProps<{
  genre: string
}>()
const appSettings = inject<AppSettings>('appSettings')
const LIMIT = appSettings?.itemsInEachGenreSection || 20
const store = useMoviesStore()
const genreShow = computed(() => store.getByGenre(genre))
const slideItems = computed<SlideItem[]>(() =>
  mapToSlideItems(genreShow.value.movies.slice(0, LIMIT) || [])
)
</script>
<template>
  <SectionTitle :title="genre">
    <IconScreen />
  </SectionTitle>
  <ThumbnailSwiper :loading="!!genreShow?.loading" :slidesPerView="7" :slides="slideItems" />
</template>
