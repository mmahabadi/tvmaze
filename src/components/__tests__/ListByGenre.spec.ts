import { mount, VueWrapper } from '@vue/test-utils'
import { describe, it, beforeEach, expect } from 'vitest'
import ListByGenre from '../ListByGenre.vue'
import SectionTitle from '@/@ui/components/SectionTitle.vue'
import ThumbnailSwiper from '@/components/thumbnail/ThumbnailSwiper.vue'
import { createPinia, setActivePinia } from 'pinia'
import { provide } from 'vue'

describe('ListByGenre', () => {
  let wrapper: VueWrapper<ListByGenre>
  provide('appSettings', {
    itemsInEachGenreSection: 20
  })

  beforeEach(() => {
    setActivePinia(createPinia())

    wrapper = mount(ListByGenre, {
      props: {
        genre: 'Action'
      },
      global: {
        stubs: {
          SectionTitle,
          ThumbnailSwiper
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders SectionTitle with correct props', () => {
    expect(wrapper.findComponent(SectionTitle).props('title')).toBe('Action')
  })

  it('renders ThumbnailSwiper with correct props', () => {
    expect(wrapper.findComponent(ThumbnailSwiper).props('loading')).toBe(false)
    expect(wrapper.findComponent(ThumbnailSwiper).props('slidesPerView')).toBe(7)
    expect(wrapper.findComponent(ThumbnailSwiper).props('slides')).toEqual([])
  })
})
