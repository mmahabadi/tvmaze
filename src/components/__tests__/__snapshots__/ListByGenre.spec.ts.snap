// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListByGenre > renders correctly 1`] = `
"<div class="my-9">
  <div class="w-full flex sm:gap-8 gap-4 items-center">
    <div class="text-teal-500"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 16 16" class="sm:w-6 sm:h-6 w-4 h-4 text-subMain" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
        <path d="M0 13a1.5 1.5 0 0 0 1.5 1.5h13A1.5 1.5 0 0 0 16 13V6a1.5 1.5 0 0 0-1.5-1.5h-13A1.5 1.5 0 0 0 0 6v7zM2 3a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 0-1h-11A.5.5 0 0 0 2 3zm2-2a.5.5 0 0 0 .5.5h7a.5.5 0 0 0 0-1h-7A.5.5 0 0 0 4 1z"></path>
      </svg></div>
    <h2 class="sm:text-xl font-bold text-lg">Action</h2>
  </div>
</div>
<div>
  <div class="swiper swiper-initialized swiper-horizontal" style="--swiper-navigation-color: rgb(20 184 166 / var(--tw-text-opacity)); --swiper-pagination-color: rgb(20 184 166 / var(--tw-text-opacity));" lazy="false">
    <div class="swiper-wrapper" style="transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"></div>
    <div class="swiper-button-prev"></div>
    <div class="swiper-button-next"></div>
    <!---->
    <!---->
  </div>
</div>"
`;
